### PR TITLE
Bug 1174405 - Check the condition again when wifi-off alarm triggers

### PR DESCRIPTION
**Pull request by timdream**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=a7deb79b862dccf9d6da0ff9a7a60bcdc0658602

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=3fc13e765c81462c781e3fa901d9eca5f867eed5

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=7f6adfba6eee995e0ff0c35eb47be9ca8294ab1d

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=9ff68815640fb98283bf1ccd3c2a490ca0dd6f3c

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=afc25e5355311cfafd030d70313d7a661c7e8ff9

File: apps/system/js/wifi.js [hunk 1]
```diff
@@ -208,15 +273,8 @@
       var lock = SettingsListener.getSettingsLock();
       // Let's quietly turn off wifi if there is no wake lock and
       // the screen is off and we are not on a power source.
-      if (!Service.query('screenEnabled') && !battery.charging) {
-        if (!this.wifiEnabled && this._wakeLockManager.isHeld) {
-          lock.set({ 'wifi.enabled': true });
-          window.addEventListener('wifi-enabled', function() {
```
Comment (timdream) [inline apps/system/js/wifi.js]: This `if` block was added in bug 979130 and should have been removed by bug 1127226.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=2071cbe499d43906c5642aeb892adab34f5bd17f

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=a5840c3ff0f1e52808e2abd4ab5414bf00b5a221

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=16ec5d3673d1e461c65bf6e8c5f36c094bd41d3e

File: apps/system/test/unit/wifi_test.js [hunk 2]
```diff
@@ -447,6 +488,46 @@ suite('WiFi > ', function() {
...
+      Wifi._alarmId = 1;
+      navigator.battery = { charging: false };
+
+      this.sinon.stub(Wifi, 'sleep');
+
+      MockMozSetMessageHandler_listeners.alarm({ data: 'wifi-off' });
+
+      assert.isTrue(Wifi.sleep.calledOnce);
+
+      navigator.battery = realBattery;
```
Comment (etiennesegonzac) [inline apps/system/test/unit/wifi_test.js]: nit: could move to a teardown

File: apps/system/test/unit/wifi_test.js [hunk 3]
```diff
@@ -447,6 +488,46 @@ suite('WiFi > ', function() {
...
+      Wifi._alarmId = 1;
+      navigator.battery = { charging: false };
+
+      this.sinon.stub(Wifi, 'sleep');
+
+      MockMozSetMessageHandler_listeners.alarm({ data: 'wifi-off' });
+
+      assert.isFalse(Wifi.sleep.calledOnce);
+
+      navigator.battery = realBattery;
```
Comment (etiennesegonzac) [inline apps/system/test/unit/wifi_test.js]: with this one

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=b56b155e522a7a8c6039ac9f7e3971c7287e489f

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=87493aa114408163de7ca4519601b693d79999b5

**Pull request merged**